src/app/digital/internal/views/components/ORGateAssembler.ts

```diff
@@ -0,0 +1,150 @@
+import {SVGDrawing} from "svg2canvas";
+
+import {V}         from "Vector";
+import {Transform} from "math/Transform";
+
+import {CircuitInternal}      from "core/internal";
+import {SelectionsManager}    from "core/internal/impl/SelectionsManager";
+import {CircuitView}          from "core/internal/view/CircuitView";
+import {PortAssembler}        from "core/internal/view/PortAssembler";
+import {QuadCurve}            from "core/internal/view/rendering/prims/QuadCurve";
+import {SVGPrim}              from "core/internal/view/rendering/prims/SVGPrim";
+import {Style}                from "core/internal/view/rendering/Style";
+import {Schema}               from "core/schema";
+import {DigitalComponentInfo} from "digital/internal/DigitalComponents";
+import {DigitalSim}           from "digital/internal/sim/DigitalSim";
+import {Assembler}            from "core/internal/view/Assembler";
+
+
+export class ORGateAssembler extends Assembler<Schema.Component> {
+    public readonly size = V(1.2, 1);
+    public readonly img: SVGDrawing;
+
+    protected readonly sim: DigitalSim;
+
+    protected portAssemblerV1: PortAssembler;
+    protected portAssemblerV2: PortAssembler;
+
+    public constructor(circuit: CircuitInternal, view: CircuitView, selections: SelectionsManager, sim: DigitalSim) {
+        super(circuit, view, selections);
+
+        this.sim = sim;
+
+        this.img = view.options.getImage("or.svg")!;
+
+        this.portAssemblerV1 = new PortAssembler(circuit, view, selections, {
+            "outputs": () => ({ origin: V(0.5, 0), dir: V(1.1, 0) }),
+            "inputs":  (index, total) => {
+                const spacing = 0.5 - this.options.defaultBorderWidth/2;
+                return { origin: V(-0.43, spacing*((total-1)/2 - index)), dir: V(-1.3, 0), stroke: {color: "black", size: 1, lineCap: "round"}
+            };
+            },
+        });
+
+        this.portAssemblerV2 = new PortAssembler(circuit, view, selections, {
```

Comment:
Please see the old code for all the math for the positioners, it *shouldnt* be this complicated, and you shouldn't need two different ones

https://github.com/OpenCircuits/OpenCircuits/blob/master/src/app/digital/models/ports/positioners/QuadraticCurvePositioner.ts#L27